bin/convert2vulyk.py

```diff
@@ -42,116 +83,173 @@ def convert_bsf_2_vulyk(text: str, bsf_markup: str) -> dict:
     """
+
+    tags_mapping = {"PERS": "ПЕРС", "ORG": "ОРГ", "LOC": "ЛОК", "MISC": "РІЗН"}
+
     bsf = parse_bsf(bsf_markup)
-    ents = [[e.id, e.tag, [[e.start_idx, e.end_idx]]] for e in bsf]
+    ents = [[e.id, tags_mapping.get(e.tag, e.tag), [[e.start_idx, e.end_idx]]] for e in bsf]
 
     idx = 0
     t_idx = 0
     s_offsets = []
     t_offsets = []
     if text:
-        for s in text.split('\n'):
+        for s in text.split("\n"):
             s_offsets.append([idx, idx + len(s)])
-            for t in s.strip().split(' '):
+            for t in s.strip().split(" "):
                 t_offsets.append([t_idx, t_idx + len(t)])
                 t_idx += len(t) + 1
 
             idx += len(s) + 1
             t_idx = idx
 
     ts = int(time.time())
-    vulyk = {"modifications": [], "equivs": [], "protocol": 1, "ctime": ts, "triggers": [], "text": text,
-              "source_files": ["ann", "txt"], "messages": [], "sentence_offsets": s_offsets, "comments": [],
-              "entities": ents, "mtime": ts, "relations": [], "token_offsets": t_offsets, "action": "getDocument",
-              "normalizations": [], "attributes": [], "events": [], "document": "", "collection": "/"}
+    vulyk = {
+        "modifications": [],
+        "equivs": [],
+        "protocol": 1,
+        "ctime": ts,
+        "triggers": [],
+        "text": text,
+        "source_files": ["ann", "txt"],
+        "messages": [],
+        "sentence_offsets": s_offsets,
+        "comments": [],
+        "entities": ents,
+        "mtime": ts,
+        "relations": [],
+        "token_offsets": t_offsets,
+        "action": "getDocument",
+        "normalizations": [],
+        "attributes": [],
+        "events": [],
+        "document": "",
+        "collection": "/",
+    }
 
     return vulyk
 
 
-def text_2_vulyk(text: str, bsf_markup: str = None) -> str:
-    """
-    Convert text string with annotations to json string for Vulyk annotation tool.
-    If annotations are provided in bsf_markup then text must be tokenized. Otherwise a raw text string can be supplied.
-    :param text: tokenzied OR raw text string
-    :param bsf_markup: string containing annotations in brat standoff format
-    :return: json string compatible with Vulyk (Brat annotation tool)
-    """
-    txt = text
-    markup = bsf_markup
-    if bsf_markup is None:
-        log.info("No markup file => processing text with Stanza to extract Named Entities.")
-        # run tokenization and NER
-        from tokenize_uk import tokenize_text
-        token_list = tokenize_text(text)
-        # we have list<paragraphs> of list<sentences> of list<tokens>
-        paragraph = ['\n'.join([' '.join(t) for t in sent]) for sent in token_list]
-        txt = '\n'.join(paragraph)  # stanza bug does not allow for double new line symbol right now
-        log.debug(txt)
+def convert(args):
+    for text in map(pathlib.Path, glob.glob(args.text_files)):
```

Comment:
I took that pattern from python doc.
https://docs.python.org/3/library/argparse.html#sub-commands

```One particularly effective way of handling sub-commands is to combine the use of the add_subparsers() method with calls to set_defaults() so that each subparser knows which Python function it should execute. For example:```

What I like about this pattern, is that you don't need to write code that says `if subcommand == "tag":` and blah-blah, especially if you have many commands (see the line 252).

But if you keen about that, I can do two wrappers, that accepts args and then passing params to the function. That way underlaying functions can be imported and used

Reply:
Oh, I see. Thank you for providing an explanation.
Split in the last commit looks good.